pkg/manager/internal.go

```diff
@@ -347,6 +352,7 @@ func (cm *controllerManager) startLeaderElection() (err error) {
 		RetryPeriod:   cm.retryPeriod,
 		Callbacks: leaderelection.LeaderCallbacks{
 			OnStartedLeading: func(_ context.Context) {
+				cm.isLeader = true
```

Comment:
Setting `isLeader` needs to be protected by the mutex. I might suggest moving it into `startLeaderElectionRunnables` and changing its name to something like `startedLeader`?

Reply:
yeap, that makes more sense. 👍